pkg/providers/common/common.go

```diff
@@ -97,3 +99,22 @@ func KubeadmConfigTemplateName(clusterName, workerNodeGroupName string, now type
 	t := now().UnixNano() / int64(time.Millisecond)
 	return fmt.Sprintf("%s-%s-template-%d", clusterName, workerNodeGroupName, t)
 }
+
+func IncrementName(name string) (string, error) {
+	re := regexp.MustCompile(`(-)(\d+)$`)
```

Comment:
`MustCompile` will panic for an invalid regex. If we want to use that, we should probably move it to a global `var`. Otherwise we should just build it normally and return an error